src/robot/parsing/model/statements.py

```diff
@@ -986,17 +991,30 @@ def condition(self):
 
     @property
     def limit(self):
-        value = self.get_value(Token.OPTION)
-        return value[len('limit='):] if value else None
+        values = self.get_values(Token.OPTION)
+        for value in values:
+            if value.startswith('limit='):
+                return value[len('limit='):]
+        return None
+
+    @property
+    def on_limit_message(self):
+        values = self.get_values(Token.OPTION)
+        for value in values:
+            if value.startswith('on_limit_message='):
+                return value[len('on_limit_message='):]
+        return None
 
     def validate(self, context):
         values = self.get_values(Token.ARGUMENT)
         if len(values) == 0:
             self.errors += ('WHILE must have a condition.',)
-        if len(values) == 2:
-            self.errors += (f"Second WHILE loop argument must be 'limit', "
+        if len(values) == 2 or len(values) == 3:
+            self.errors += (
+                            f"WHILE loop arguments must be 'limit' "
+                            f"or 'on_limit_message', "
                             f"got '{values[1]}'.",)
```

Comment:
Reporting all possible options here if WHILE has more than one condition, or an invalid option, can get a bit complicated. For example, the above error message shows only the second argument, not possible third. I'm not sure how much effort should be spent on this. I'd be fine with something simple like below, but a better error listing all possible options certainly would have benefits.

```python
if len(values) > 1:
    self.errors +) ('WHILE cannot have more than one condition, got {seq2str(values)}.',)
```